backend/src/managers/UserManager.ts

```diff
@@ -127,16 +128,30 @@ export default class UserManager {
     }
 
     async getUserStats(forUserId: number): Promise<UserStats> {
+        if (this.userStatsCache.has(forUserId)) {
+            return this.userStatsCache.get(forUserId);
+        }
+
         const unreadComments = await this.userRepository.getUserUnreadComments(forUserId);
         const notifications = await this.notificationManager.getNotificationsCounts(forUserId);
 
-        return {
+        const stats = {
             notifications,
             watch: {
                 comments: unreadComments,
                 posts: 0
             }
         };
+
+        this.userStatsCache.set(forUserId, stats);
+        return stats;
+    }
+
+    deleteUserStatsCache(forUserId: number) {
+        this.userStatsCache.delete(forUserId);
+    }
+    clearUserStatsCache() {
```

Comment:
this is called on each comment posted on site, not sure if having this cache would be beneficial

Reply:
It would. The cache is cheap. Currently the **average** time interval between comments is 1.5 minutes, meaning ≈2 cache hits per user (assuming 30s status update), more in practice.

One future improvement would be to have an intelligent cache eviction behavior that depends on whether the comment is posted in a post with low number of comments/watches. In such case only affected users could be evicted from the cache.